### ISSUE TITLE
Too many old logs break the addon

### ISSUE THREAD

**Issue by BadgerCode** [BadgerCode/tttdamagelogs#8]
> When you reach max capacity, you can only view up to what's stored and no longer records/saves them
> I have to unclog it by clearing them from the sv.db

The addon automatically deletes logs older than 61 days (set in the config by the property `Damagelog.LogDays`).

I'm not sure exactly what the limit is.
If this is the same problem I used to have, once you have too many logs, trying to browse old logs breaks the in-game UI.

**Comment by BadgerCode** [comment]
This is partially caused by the way the damage logs addon generates IDs for old log rows in the database.

# Steps to reproduce
* Have 100 old logs in your database, with IDs 1-100
* Set the date for log ID 1 to 62 days ago
* Set the date for logs IDs 2-100 to within the past 60 days
* Set the max age for old logs to 61 days
* Play a round on the server

# Expected Results
* Log ID 1 is deleted, as it is older than the maximum age allowed
* The round you play is added as a new row with ID 101

# Actual Results
* Log ID 1 is deleted, as it is older than the maximum age allowed
* The round you play is not inserted to the database due to the following error:
    * `UNIQUE constraint failed: damagelog_oldlogs_v3.id`


# Explanation
The insert fails because the code tries to insert a row with ID 100.
After log ID 1 is deleted, there are 99 rows in the database with IDs 2-100.
When the addon tries to insert a new row, it runs the following code to get an ID:
```lua
local function GetLogsCount_SQLite()
    return sql.QueryValue("SELECT COUNT(id) FROM damagelog_oldlogs_v3") -- This returns 99 as there are 99 rows
end

-- ... later on

local id = GetLogsCount_SQLite() + 1 -- This equals 100 (99 + 1)
```

This code returns a value of 100 for the ID, which conflicts with the existing row with ID 100.


# Fix

Change
`SELECT COUNT(id) FROM damagelog_oldlogs_v3`

To
`SELECT MAX(id) FROM damagelog_oldlogs_v3`

**Comment by BadgerCode** [comment]
There is also a potential limit when transferring the old logs information to the clientside.

In `lua\damagelogs\server\oldlogs.lua`, line 207:
`net.WriteTable(Damagelog.OldLogsDays)`

This has a limit of 64KB.
Need to check if this is an issue.

Using JSON & compression would be better.

**Comment by BadgerCode** [comment]
Fixed in #10 
Will be released in #9

**Comment by BadgerCode** [comment]
Potentially fixes https://github.com/Tommy228/tttdamagelogs/issues/357 too
